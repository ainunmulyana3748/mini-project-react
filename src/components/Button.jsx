import React from "react";

const Button = ({
  children,
  variant = "primary",
  onclick,
  disabled,
  ...props
}) => {
  const base = "px-4 py-2 rounded font-medium";
  const variants = {
    primary:
      "bg-black text-white hover:bg-zinc-900 focus:ring-black disabled:opacity-50 cursor-pointer",
    secondary:
      "bg-white text-black border border-gray-300 hover:bg-gray-100 focus:ring-gray-400 disabled:opacity-50 cursor-pointer",
  };

  return (
    <button
      disabled={disabled}
      className={`${base} ${variants[variant]}`}
      onClick={onclick}
      {...props}
    >
      {children}
    </button>
  );
};

export default Button;
