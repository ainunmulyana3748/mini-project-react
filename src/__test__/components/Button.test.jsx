import { render, screen, fireEvent } from "@testing-library/react";
import Button from "../../components/Button";

describe("Button Component", () => {
  test("render children dengan benar", () => {
    render(<Button>Click Me</Button>);
    expect(screen.getByText("Click Me")).toBeInTheDocument();
  });

  test("render dengan variant primary secara default", () => {
    render(<Button>Primary</Button>);
    expect(screen.getByText("Primary")).toHaveClass("bg-black");
  });

  test("render dengan variant secondary", () => {
    render(<Button variant="secondary">Secondary</Button>);
    expect(screen.getByText("Secondary")).toHaveClass("bg-white");
  });

  test("render dengan variant danger", () => {
    render(<Button variant="danger">Danger</Button>);
    expect(screen.getByText("Danger")).toHaveClass("bg-red-500");
  });

  test("onClick dijalankan saat button diklik", () => {
    const handleClick = jest.fn();
    render(<Button onclick={handleClick}>Click</Button>);
    fireEvent.click(screen.getByText("Click"));
    expect(handleClick).toHaveBeenCalled();
  });

  test("button disabled tidak bisa diklik", () => {
    const handleClick = jest.fn();
    render(
      <Button onclick={handleClick} disabled>
        Disabled
      </Button>
    );
    const btn = screen.getByText("Disabled");
    expect(btn).toBeDisabled();
    fireEvent.click(btn);
    expect(handleClick).not.toHaveBeenCalled();
  });

  test("snapshot button", () => {
    const { container } = render(<Button>Snapshot</Button>);
    expect(container).toMatchSnapshot();
  });
});
