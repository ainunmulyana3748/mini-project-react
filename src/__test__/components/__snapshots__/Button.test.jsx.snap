// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component snapshot button 1`] = `
<div>
  <button
    class="px-4 py-2 rounded font-medium bg-black text-white hover:bg-zinc-900 focus:ring-black disabled:opacity-50 cursor-pointer"
  >
    Snapshot
  </button>
</div>
`;
